test/integration_helper.rb

```diff
@@ -0,0 +1,114 @@
+# frozen_string_literal: true
+
+module IntegrationHelper
+  def spans
+    # Wait for spans that haven't been written yet
+    sleep 1
+    File.readlines(SPANS_FILE_PATH).map do |line|
+      JSON.parse(line)
+    end
+  end
+
+  def root_spans
+    spans.select do |span|
+      span["parentSpanId"].nil?
+    end
+  end
+
+  def root_span
+    return if root_spans.length != 1
+
+    root_spans.first
+  end
+
+  def root_span!
+    raise "There is no root span" if root_spans.length.zero?
+    raise "There is more than one root span" if root_spans.length > 1
+
+    root_span
+  end
+
+  # Given a span ID, returns the span, or `nil` if the span does not exist.
+  def span(id)
+    spans.find do |span|
+      span["spanId"] == id
+    end
+  end
+
+  # Given a parent span, returns its child spans.
+  def child_spans(parent)
+    spans.select do |span|
+      span["parentSpanId"] == parent["spanId"]
+    end
+  end
+
+  # Given a child span, returns its parent span, or `nil` if a root span is
+  # passed.
+  def parent_span(child)
+    span(child["parentSpanId"])
+  end
+
+  # Returns whether the child span is a transitive child of the parent span.
+  def child_span_of?(parent, child)
+    while child && child["parentSpanId"]
+      return true if child["parentSpanId"] == parent["spanId"]
+
+      child = span(child["parentSpanId"])
+    end
+
+    false
+  end
+
+  def expect_http_root_span(name)
```

Comment:
I'll keep all these instrumentation-specific helpers here until we have all the apps in place, because I can see all of them being reused. Even the Redis ones.